python/dgl/graphbolt/base.py

```diff
@@ -38,12 +38,30 @@
     "CSCFormatBase",
     "seed",
     "seed_type_str_to_ntypes",
+    "get_host_to_device_uva_stream",
+    "get_device_to_host_uva_stream",
 ]
 
 CANONICAL_ETYPE_DELIMITER = ":"
 ORIGINAL_EDGE_ID = "_ORIGINAL_EDGE_ID"
 
 
+# There needs to be a single instance of the uva_stream, if it is created
+# multiple times, it leads to multiple CUDA memory pools and memory leaks.
+def get_host_to_device_uva_stream():
```

Comment:
I know python is a very flexible language, but this is probably less error prone, recommended by chatgpt, not sure whether it works or not.

```
def get_singleton_host_to_device_uva_stream():
    _stream = None

    def get_stream():
        nonlocal _stream
        if _stream is None:
            _stream = torch.cuda.Stream(priority=-1)
        return _stream

    return get_stream()
```

Reply:
What is wrong with current approach, this was actually recommended on stackoverflow.

Reply:
I see, just looks uncomfortable, since you assign a variable to a method. Maybe I am just uncomfortable of python :)

Reply:
https://stackoverflow.com/a/279586/2904531

Reply:
But ChatGPT suggestion works as well.

Reply:
I see, I hate python :)

Reply:
Why they don't add a static keyword is unbeknownst to me.